### PR TITLE
Add optional `default` parameter to `select_first`

### PR DESCRIPTION
**Pull request by jdidion**
### Checklist
- [x] Pull request details were added to CHANGELOG.md
- [x] Valid examples WDL's were added or updated to the SPEC.md (see the [guide](https://github.com/openwdl/wdl-tests/blob/main/docs/MarkdownTests.md) on writing markdown tests)

### CODE REVIEW

**Review by vsmalladi — APPROVED** [review]
LGTM

File: SPEC.md [hunk 1]
```diff
@@ -9571,9 +9573,11 @@ workflow test_select_first {
...
   output {
-    # both of these statements evaluate to 5
-    Int five1 = select_first([maybe_five, maybe_four_but_is_not, maybe_three])
-    Int five2 = select_first([maybe_four_but_is_not, maybe_five, maybe_three])
+    # all of these statements evaluate to 5
+    Int fiveA = select_first([maybe_five, maybe_four_but_is_not, maybe_three])
+    Int fiveB = select_first([maybe_four_but_is_not, maybe_five, maybe_three])
+    Int fiveC = select_first([], 5)
+    Int fiveD = select_first([None], 5)
   }
```
Comment (wleepang) [inline SPEC.md:9743]: Should error cases be represented here as well?

Also, what is the expected behavior if handling `Array[Array[T]]` where the first element is `[]`?

Comment (jdidion) [reply SPEC.md:9743]: The error cases are in the subsequent examples.

As to your second question, I think you're asking about the following

```
Array[Array[Int]] a = [[], [1, 2, 3]]
Array[Int] a_first = select_first(a)
```

in which case `a_first == []`, i.e., empty array is not the same as undefined.

**Review by illusional — APPROVED** [review]
LGTM!

**Pull request merged**